File: external/Java.Interop/src/Java.Interop/Tests/Java.Interop/CallNonvirtualDerived2.cs
using System;

using Java.Interop;

namespace Java.InteropTests
{
	[JniTypeInfo (CallNonvirtualDerived2.JniTypeName)]
	public class CallNonvirtualDerived2 : CallNonvirtualDerived
	{
		internal new const string JniTypeName = "com/xamarin/interop/CallNonvirtualDerived2";

		public CallNonvirtualDerived2 ()
		{
		}
	}
}

File: external/Java.Interop/src/Java.Interop/Tests/java/com/xamarin/interop/CallNonvirtualDerived2.java
package com.xamarin.interop;

import java.util.ArrayList;

import com.xamarin.java_interop.GCUserPeerable;

public class CallNonvirtualDerived2
		extends CallNonvirtualDerived
		implements GCUserPeerable
{
	ArrayList<Object>       managedReferences     = new ArrayList<Object>();

	public CallNonvirtualDerived2 () {
		com.xamarin.java_interop.ManagedPeer.runConstructor (
				CallNonvirtualDerived2.class,
				this,
				"Java.InteropTests.CallNonvirtualDerived2, Java.Interop-Tests, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null",
				""
		);
	}

	public void jiAddManagedReference (java.lang.Object obj)
	{
		managedReferences.add (obj);
	}

	public void jiClearManagedReferences ()
	{
		managedReferences.clear ();
	}
}
